app/controllers/workflow/pages_controller.rb

```diff
@@ -5,128 +5,187 @@ class Workflow::PagesController < ApplicationController
+        element = workflow_approver.split(",")
+        target_user = SS::User.find(element[1]) rescue nil
+        if target_user
+          email_blank_users.push(target_user.name) if target_user.email_disabled?
+        end
+      end
+    else
       current_level = @item.workflow_current_level
       current_workflow_approvers = @item.workflow_approvers_at(current_level)
       current_workflow_approvers.each do |workflow_approver|
-        args = { f_uid: @cur_user._id, t_uid: workflow_approver[:user_id],
-                 site: @cur_site, page: @item,
-                 url: params[:url], comment: params[:workflow_comment] }
-        Workflow::Mailer.request_mail(args).deliver_now
+        target_user = SS::User.find(workflow_approver[:user_id]) rescue nil
+        if target_user
+          email_blank_users.push(target_user.name) if target_user.email_disabled?
+        end
       end
-
-      @item.set_workflow_approver_state_to_request
-      @item.update
+      target_user = nil
+      target_user_id = @item.workflow_user_id || @cur_user._id
+      target_user = SS::User.find(target_user_id) rescue nil
+      if target_user
+        email_blank_users.push(target_user.name) if target_user.email_disabled?
+      end
+    end
+    email_blank_users.uniq!
+    email_blank_users.sort!
+    return nil if email_blank_users.blank?
+    message = t("errors.messages.user_email_blank")
+    email_blank_users.each do |email_blank_user|
+      message += "\n#{email_blank_user}"
     end
+    message
+  end
 
   public
-    def request_update
-      raise "403" unless @item.allowed?(:edit, @cur_user)
 
-      @item.workflow_user_id = @cur_user._id
-      @item.workflow_state   = "request"
-      @item.workflow_comment = params[:workflow_comment]
-      @item.workflow_approvers = params[:workflow_approvers]
-      @item.workflow_required_counts = params[:workflow_required_counts]
+  def request_update
+    raise "403" unless @item.allowed?(:edit, @cur_user)
 
-      if @item.update
-        request_approval
-        render json: { workflow_state: @item.workflow_state }
-      else
-        render json: @item.errors.full_messages, status: :unprocessable_entity
+    if params[:forced_update_option] == "false"
+      if message = workflow_alert
+        render json: { workflow_alert: message }
+        return
       end
+    end
 
+    @item.workflow_user_id = @cur_user._id
+    @item.workflow_state   = "request"
+    @item.workflow_comment = params[:workflow_comment]
+    @item.workflow_approvers = params[:workflow_approvers]
+    @item.workflow_required_counts = params[:workflow_required_counts]
+
+    if @item.update
+      request_approval
+      render json: { workflow_state: @item.workflow_state }
+    else
+      render json: @item.errors.full_messages, status: :unprocessable_entity
     end
+  end
 
-    def approve_update
-      raise "403" unless @item.allowed?(:approve, @cur_user)
+  def approve_update
```

Comment:
__[rubocop]__
* Perceived complexity for approve_update is too high. [15/14]

<!-- Comment by [SideCI](https://sideci.com) -->